discord/poll.py

```diff
@@ -0,0 +1,641 @@
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+
+from typing import Optional, List, TYPE_CHECKING, Union, AsyncIterator
+
+import datetime
+
+from .enums import PollLayoutType, try_enum
+from . import utils
+from .emoji import PartialEmoji, Emoji
+from .user import User
+from .object import Object
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .message import Message
+    from .abc import Snowflake
+    from .state import ConnectionState
+    from .member import Member
+
+    from .types.poll import (
+        PollCreate as PollCreatePayload,
+        PollMedia as PollMediaPayload,
+        PollAnswerCount as PollAnswerCountPayload,
+        Poll as PollPayload,
+        PollAnswerWithID as PollAnswerWithIDPayload,
+        PollResult as PollResultPayload,
+    )
+
+
+__all__ = (
+    'Poll',
+    'PollAnswer',
+    'PollMedia',
+)
+
+MISSING = utils.MISSING
+PollMediaEmoji = Union[PartialEmoji, Emoji, str]
+
+
+class PollMedia:
+    """Represents the poll media for a poll item.
+
+    .. container:: operations
+
+        .. describe:: str(x)
+
+            Returns this Poll Media\'s text.
+
+    Attributes
+    ----------
+    text: :class:`str`
+        The displayed text
+    emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`, :class:`str`]]
+        The attached emoji for this media. This will always be ignored for a poll
+        question media.
+    """
+
+    __slots__ = ('text', 'emoji')
+
+    def __init__(self, /, text: str, emoji: Optional[PollMediaEmoji] = None) -> None:
+        self.text: str = text
+        self.emoji: Optional[PollMediaEmoji] = emoji
+
+    def __str__(self) -> str:
+        return self.text
+
+    def __repr__(self) -> str:
+        return f'<PollMedia text={self.text!r} emoji={self.emoji!r}>'
+
+    def to_dict(self) -> PollMediaPayload:
+        """Returns an API valid payload for this instance."""
+
+        payload: PollMediaPayload = {'text': self.text}
+
+        if self.emoji:
+            if isinstance(self.emoji, Emoji):
+                payload['emoji'] = {'name': self.emoji.name, 'id': self.emoji.id}
+
+            elif isinstance(self.emoji, PartialEmoji):
+                payload['emoji'] = self.emoji.to_dict()
+
+            else:
+                payload['emoji'] = {'name': str(self.emoji)}  # type: ignore
```

Comment:
```suggestion
        if isinstance(self.emoji, str):
            payload['emoji'] = {"id": None, "name": self.emoji}
        elif self.emoji is not None:
            payload['emoji'] = {"id": self.emoji.id, "name": self.emoji.name}
```

This can be refworked to remove the type-ignore.